### PR TITLE
Adding support for HAProxy JSON logs

### PR DESCRIPTION
**Pull request by perangel**
* Update haproxy.tmpl to use JSON as log-format
    * Update logstash.conf to filter JSON content using "message" field

### CODE REVIEW

File: terraform/aws/s3/etc_logstash.conf [hunk 1]
```diff
@@ -28,6 +28,12 @@ filter {
   }
 }
 
+filter {
+  json {
+    source => "message"
```
Comment (pmbauer) [inline terraform/aws/s3/etc_logstash.conf]: https://github.com/logstash-plugins/logstash-filter-json/blob/master/lib/logstash/filters/json.rb#L76
we might want to flag skip_on_invalid_json - many of the "message" fields won't contain json with this

Comment (pmbauer) [inline terraform/aws/s3/etc_logstash.conf]: :( this flag isn't released yet, current version of logstash does not support.
this does mean every syslog event that does not have json in the message field will raise a warning and get the event tagged with a json parse failure

**Comment by pmbauer** [comment]
lgtm; I expect a 15-20 min gap in some logs, depending on how long it takes for logspout and rsyslog to realize their old logstash connection has gone stale (based on past testing).
proposed prod deploy sequence
```
# upload s3 template
make update CMD='run_terraform apply'

# restart logstash service
# wait for logspout and rsyslog to start sending logs again
# this will result in haproxy syslogjson events being tagged for a bit with json parse errors, but otherwise readable

make update_proxy
```

**Comment by perangel** [comment]
I've noticed that Terraform doesn't pick up on the logstash.conf being different, so I've had to delete the file in S3 before running the `make update`. Otherwise the file doesn't change (or I was doing something wrong)

**Comment by pmbauer** [comment]
right; I forgot that resource has to get `terraform taint`'d first

**Pull request merged**